apps/admin/backend/src/store.ts

```diff
@@ -651,17 +714,44 @@ export class Store {
     ) as Array<{
       id: Id;
       ballotId: string;
-      data: string;
+      ballotStyleId: string;
+      ballotType: string;
+      batchId: string;
+      batchLabel: string;
+      precinctId: string;
+      scannerId: string;
+      sheetNumber: number | null;
+      votes: string;
       createdAt: Iso8601Timestamp;
     }>;
 
-    return entries.map((entry) => ({
-      id: entry.id,
-      ballotId: entry.ballotId,
-      electionId,
-      data: entry.data,
-      createdAt: convertSqliteTimestampToIso8601(entry.createdAt),
-    }));
+    const fileMode = this.getCurrentCvrFileModeForElection(electionId);
+    if (fileMode === Admin.CvrFileMode.Unlocked) return [];
+    const isTestMode = fileMode === Admin.CvrFileMode.Test;
+
+    return entries.map((entry) => {
+      const castVoteRecordLegacyMetadata: CastVoteRecord = {
+        _precinctId: entry.precinctId,
+        _scannerId: entry.scannerId,
+        _batchId: entry.batchId,
+        _batchLabel: entry.batchLabel,
+        _ballotStyleId: entry.ballotStyleId,
+        _ballotType: cvrBallotTypeToLegacyBallotType(
+          entry.ballotType as CVR.vxBallotType
+        ),
+        _testBallot: isTestMode,
+      };
+      return {
+        id: entry.id,
+        ballotId: entry.ballotId,
+        electionId,
+        data: JSON.stringify({
+          ...castVoteRecordLegacyMetadata,
+          ...JSON.parse(entry.votes),
+        }),
+        createdAt: convertSqliteTimestampToIso8601(entry.createdAt),
+      };
+    });
```

Comment:
You'll see in the performance testing document that this conversion is slowing us down quite a bit, but it's temporary.